src/Impl/GuzzleHttpRequester.php

```diff
@@ -24,11 +24,19 @@ public function __construct(?ClientInterface $httpClient = null)
         $this->httpClient = $httpClient ?? new GuzzleHttpClient();
     }
 
-    public function performIngestion(
-        UriInterface $endpointUri,
-        array $headers,
-        string $json
-    ): array {
+    public function isIngestionServiceAvailable(UriInterface $endpointUri): bool {
+        try {
+            $request = new Request('GET', $endpointUri);
+            $response = $this->httpClient->sendRequest($request);
+            return $this->parseHealthCheckResponse($response);
+        } catch (ClientExceptionInterface $e) {
+            throw new StreamxClientException(
+                sprintf('HealthCheck GET request with URI: %s failed due to HTTP client error', $endpointUri),
+                $e);
+        }
+    }
+
+    public function performIngestion(UriInterface $endpointUri, array $headers, string $json): array {
```

Comment:
Put params in single line for all (now 3) methods:
 - isIngestionServiceAvailable (new)
 - performIngestion (existing)
 - fetchSchema (existing)